pennylane_lightning/lightning_qubit/lightning_qubit2.py

```diff
@@ -0,0 +1,511 @@
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Defaults to ``None`` if not specified. Setting
+            to ``None`` results in computing statistics like expectation values and
+            variances analytically.
+        seed (Union[str, None, int, array_like[int], SeedSequence, BitGenerator, Generator]): A
+            seed-like parameter matching that of ``seed`` for ``numpy.random.default_rng``, or
+            a request to seed from numpy's global random number generator.
+            The default, ``seed="global"`` pulls a seed from NumPy's global generator. ``seed=None``
+            will pull a seed from the OS entropy.
+        mcmc (bool): Determine whether to use the approximate Markov Chain Monte Carlo
+            sampling method when generating samples.
+        kernel_name (str): name of transition kernel. The current version supports
+            two kernels: ``"Local"`` and ``"NonZeroRandom"``.
+            The local kernel conducts a bit-flip local transition between states.
+            The local kernel generates a random qubit site and then generates a random
+            number to determine the new bit at that qubit site. The ``"NonZeroRandom"`` kernel
+            randomly transits between states that have nonzero probability.
+        num_burnin (int): number of steps that will be dropped. Increasing this value will
+            result in a closer approximation but increased runtime.
+        batch_obs (bool): Determine whether we process observables in parallel when
+            computing the jacobian. This value is only relevant when the lightning
+            qubit is built with OpenMP.
+    """
+
+    _device_options = ("rng", "c_dtype", "batch_obs", "mcmc", "kernel_name", "num_burnin")
+    _CPP_BINARY_AVAILABLE = LQ_CPP_BINARY_AVAILABLE
+    _new_API = True
+
+    operations = _operations
+    """The names of the supported operations."""
+
+    observables = _observables
+    """The names of the supported observables."""
+
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        wires,
+        *,
+        c_dtype=np.complex128,
+        shots=None,
+        seed="global",
+        mcmc=False,
+        kernel_name="Local",
+        num_burnin=100,
+        batch_obs=False,
+    ):
+        if not LQ_CPP_BINARY_AVAILABLE:
+            raise ImportError(
+                "Pre-compiled binaries for lightning.qubit are not available. "
+                "To manually compile from source, follow the instructions at "
+                "https://pennylane-lightning.readthedocs.io/en/latest/installation.html."
+            )
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._statevector = LightningStateVector(num_wires=len(self.wires), dtype=c_dtype)
+        seed = np.random.randint(0, high=10000000) if seed == "global" else seed
+        self._rng = np.random.default_rng(seed)
```

Comment:
We should revisit this at some point, because we may end up using the C++ RNGs for sampling which will make this moot.

Reply:
Yes, this is here for now because Christina and I weren't sure if samples are generated in the Python or C++ layer. I think it's best to leave it for now and remove it (or not) in the shots PR.